### PR TITLE
get route params instead of using trait

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug/Hotfix?   | no
| Refactoring?  | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no 
| Tests pass?   | yes

Get route parameters instead of using a trait inside an entity

### CODE REVIEW

**Comment by sensorario** [comment]
I think that thanks to this PR it is possible to mark as `deprecate` method `customQueryStringValues()`. This is not a feature, this is very nice improvement!!!

**Comment by AlessandroMinoccheri** [comment]
I think that this PR can solve the big problem adding customQueryStringValues trait to every entities that is used in this bundle.
Actually if you forget to add customQueryStringValues into entities you don't have control on It and code will be broken.

File: src/Mado/QueryBundle/Repositories/BaseRepository.php [hunk 1]
```diff
@@ -183,6 +183,7 @@ public function setQueryOptionsFromRequestWithCustomFilter(Request $request = nu
 
         $this->queryOptions = QueryBuilderOptions::fromArray([
             '_route' => $request->attributes->get('_route'),
+            '_route_params' => $request->attributes->get('_route_params'),
```
[the remark this answers is not in the document]
Comment (AlessandroMinoccheri) [reply src/Mado/QueryBundle/Repositories/BaseRepository.php]: throw an exception or use an empty array to manage this problem?

File: tests/Mado/QueryBundle/Services/RouterTest.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            '_route_params' => [$routeParamsKey => $routeParamsValue]
+        ]);
+
+        $this->router = new Router();
+        $route = $this->router->createRouter($queryBuilderOptions, $routeName);
+        $routeParams = $route->getParameters();
+
+        $this->assertInstanceOf('Hateoas\Configuration\Route', $route);
+        $this->assertEquals($routeName, $route->getName());
+        $this->assertTrue(array_key_exists($routeParamsKey, $routeParams));
```
[the remark this answers is not in the document]
Comment (AlessandroMinoccheri) [reply tests/Mado/QueryBundle/Services/RouterTest.php]: The important is the key not the value

File: src/Mado/QueryBundle/Repositories/BaseRepository.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (sensorario) [reply src/Mado/QueryBundle/Repositories/BaseRepository.php]: `$request->attributes->get('_route_params')` -> `$request->attributes->get('_route_params', [])`

[the remark this answers is not in the document]
Comment (sensorario) [reply src/Mado/QueryBundle/Repositories/BaseRepository.php]: Better the empty array

**Pull request merged**